ContractsApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs

```diff
@@ -325,5 +329,44 @@ public async Task PatchContractSuccessfullyUpdatesAContract()
 
             await _dbFixture.DynamoDbContext.DeleteAsync<ContractDb>(contractId).ConfigureAwait(false);
         }
+
+        [Fact]
+        public async Task ApprovalStatusStoredAndRetrievedCorrectly()
+        {
+            // Arrange
+            var contract = _fixture.Build<ContractDb>()
+                .With(x => x.VersionNumber, (int?)null)
+                .With(x => x.ApprovalStatus, ApprovalStatus.PendingApproval)
+                .Create();
+
+            // Act
+            await _classUnderTest.PostNewContractAsync(contract).ConfigureAwait(false);
+
+            // Assert
+            var storedContract = await _dbFixture.DynamoDbContext.LoadAsync<ContractDb>(contract.Id).ConfigureAwait(false);
+            storedContract.Should().NotBeNull();
+            storedContract.ApprovalStatus.Should().Be(ApprovalStatus.PendingApproval);
+
+            // Cleanup
+            await _dbFixture.DynamoDbContext.DeleteAsync<ContractDb>(contract.Id).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public void ApprovalStatusStoredAsStringInDynamoDb()
+        {
+            // Arrange
+            var contract = _fixture.Build<ContractDb>()
+                .With(x => x.VersionNumber, (int?)null)
+                .With(x => x.ApprovalStatus, ApprovalStatus.PendingApproval)
+                .Create();
+
+            //Act
+            var document = _dbFixture.DynamoDbContext.ToDocument(contract);
+
+            //Assert
+            var approvalStatusAttribute= document["approvalStatus"];
+            approvalStatusAttribute.Should().NotBeNull();
+            approvalStatusAttribute.AsString().Should().Be("PendingApproval");
```

Comment:
Looking at the documentation, it's the test that does the data type or format creation:

![image](https://github.com/user-attachments/assets/92e7fc38-0f9a-4aab-b6b3-ca2fd3e19df6)


Given the conversion of whatever we have is forced by the test itself to be into a string,
how can we be sure that the `approvalStatusAttribute` would result in an integer without the `.AsString()` call done by the test.

What if you do:

![image](https://github.com/user-attachments/assets/d9a438d1-7cbb-4570-8dd6-a77d2bf65e0f)


```cs
approvalStatusAttribute.Value.Should().Be("PendingApproval");
```
Does it still pass?
If no, is it because it implicitly converts to a stringified number? :thinking:

Reply:
![Screenshot 2024-10-17 at 13 45 06](https://github.com/user-attachments/assets/31b10e1e-d40e-440a-868c-e6a130a48a52)
All good

Reply:
Did you give it a go trying out the `.Value` approach:
`approvalStatusAttribute.Value.Should().Be("PendingApproval");` 

Instead of `.ToString()`?
`approvalStatusAttribute.ToString().Should().Be("PendingApproval");`

Because the `.ToString()` is just another converter. :thinking: 
Or is there no `.Value` attribute on that Primitive object?

Reply:
I did indeed and saw that the approvalStatusAttribute is stored in JSON.
The ToString() method is to convert the PendingApproval from JSON to a string in order to compare it.
I also printed the variable document and used an external JSON converter to verify that the approvalStatus is living as a string within the dynamoDb.
Please let me know if there are further checks I should be undertaking and how can I perform them.

Reply:
If that's what it is, then it was all good as it was.
So odd it's giving you JSON... not mentioned in the documentation I found :thinking: